### PR TITLE
explain error when publishing in bundle mode

### PR DESCRIPTION
**Pull request by JerryKwan**
explain the error generated when publish multiple modules in bundle mode

## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/move-language/move/blob/main/CONTRIBUTING.md#developer-workflow)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

### CODE REVIEW

**Comment by JerryKwan** [comment]
@meng-xu-cs Would you please help to review? thank you.

**Comment by JerryKwan** [comment]
@meng-xu-cs 
This PR is related to issue https://github.com/move-language/move/issues/438

File: language/tools/move-cli/tests/sandbox_tests/cyclic_multi_module_publish/args.exp [hunk 1]
```diff
@@ -6,3 +6,4 @@ Wrote 175 bytes of module ID's and code
 Command `-p p2 sandbox publish --bundle --override-ordering A --override-ordering C -v`:
 Found 3 modules
 Invalid multi-module publishing: VMError with status INVALID_FRIEND_DECL_WITH_MODULES_IN_DEPENDENCIES at location Module ModuleId { address: 00000000000000000000000000000003, name: Identifier("C") } and message At least one module, 00000000000000000000000000000003::A, appears in both the dependency set and the friend set
+Publishing failed with unexpected error INVALID_FRIEND_DECL_WITH_MODULES_IN_DEPENDENCIES
```
Comment (meng-xu-cs) [inline language/tools/move-cli/tests/sandbox_tests/cyclic_multi_module_publish/args.exp:9]: Would you mind leaving a TODO item to explain this?

File: language/tools/move-cli/src/sandbox/commands/publish.rs [hunk 2]
```diff
@@ -115,8 +116,16 @@ pub fn publish(
...
-                        // TODO (mengxu): explain publish errors in multi-module publishing
                         println!("Invalid multi-module publishing: {}", err);
+                        if let Location::Module(module_id) = err.location() {
+                            // find the module where error occures and explain
+                            if let Some(unit) = modules_to_publish
+                                .into_iter()
+                                .find(|&x| x.unit.name().as_str() == module_id.name().as_str())
+                            {
+                                explain_publish_error(err, state, unit)?
+                            }
```
Comment (meng-xu-cs) [inline language/tools/move-cli/src/sandbox/commands/publish.rs:127]: I would probably leave a println on the else path as well, with a message like "Unable to locate the module in the multi-module publishing error"

**Review by meng-xu-cs — COMMENTED** [review]
Mostly LGTM with minor comments.

File: language/tools/move-cli/tests/sandbox_tests/cyclic_multi_module_publish/args.exp (hunk 1 above)
Comment (JerryKwan) [reply language/tools/move-cli/tests/sandbox_tests/cyclic_multi_module_publish/args.exp:9]: This was generated by the explain_publish_error() function.
I did not remove the line 
```
println!("Invalid multi-module publishing: {}", err); 
```
in order to have the other test cases run smoothly

**Comment by JerryKwan** [comment]
@meng-xu-cs add more exception message as suggested, please help to review.

**Review by meng-xu-cs — APPROVED** [review]

**Pull request merged**